### ISSUE TITLE
window11install flash_attn train issue

### ISSUE THREAD

**Issue by partcompany** [Dao-AILab/flash-attention#1550]
I installed the window version flash_attn-2.7.4-post1+cu124torch2.5.1cxx11abiFALSE-cp311-cp311-win_amd64.whl during training. After multiple rounds of training, the model did not converge. After uninstalling flash_attn, I trained it normally using scaled_otsproduct_mattention, How to solve this problem? Here is my configuration，:

`Transferred 693/799 items from pretrained weights
Ultralytics 8.3.63 🚀 Python-3.11.11 torch-2.5.1+cu124 CUDA:0 (NVIDIA GeForce RTX 4090, 24563MiB)
engine\trainer: task=detect, mode=train, model=D:\LYS\pycharmcode\yolov12-main\ultralytics\cfg\models\v12\yolov12m.yaml, data=D:\LYS\pycharmcode\yolov12-main\dataset\data.yaml, epochs=200, time=None, patience=100, batch=32, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=4, project=runs/train, name=train4, exist_ok=False, pretrained=yolov12m.pt, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.0, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.1, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\train\train4
Overriding model.yaml nc=80 with nc=5

                   from  n    params  module                                       arguments                     
  0                  -1  1      1856  ultralytics.nn.modules.conv.Conv             [3, 64, 3, 2]                 
  1                  -1  1     37120  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2, 1, 2]         
  2                  -1  1    111872  ultralytics.nn.modules.block.C3k2            [128, 256, 1, True, 0.25]     
  3                  -1  1    147968  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2, 1, 4]        
  4                  -1  1    444928  ultralytics.nn.modules.block.C3k2            [256, 512, 1, True, 0.25]     
  5                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  6                  -1  2   2664960  ultralytics.nn.modules.block.A2C2f           [512, 512, 2, True, 4]        
  7                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  8                  -1  2   2664960  ultralytics.nn.modules.block.A2C2f           [512, 512, 2, True, 1]        
  9                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 10             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 11                  -1  1   1248768  ultralytics.nn.modules.block.A2C2f           [1024, 512, 1, False, -1]     
 12                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 13             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 14                  -1  1    378624  ultralytics.nn.modules.block.A2C2f           [1024, 256, 1, False, -1]     
 15                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
 16            [-1, 11]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 17                  -1  1   1183232  ultralytics.nn.modules.block.A2C2f           [768, 512, 1, False, -1]      
 18                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
 19             [-1, 8]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 20                  -1  1   1642496  ultralytics.nn.modules.block.C3k2            [1024, 512, 1, True]          
 21        [14, 17, 20]  1   1414879  ultralytics.nn.modules.head.Detect           [5, [256, 512, 512]]          
YOLOv12m summary: 533 layers, 19,612,959 parameters, 19,612,943 gradients, 60.1 GFLOPs

Transferred 793/799 items from pretrained weights
Freezing layer 'model.21.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks skipped ⚠️. Unable to load YOLO11n for AMP checks due to possible Ultralytics package modifications. Setting 'amp=True'. If you experience zero-mAP or NaN losses you can disable AMP with amp=False.
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154719435_10.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154722049_8.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154724552_6.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154727009_5.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154727009_6.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154727009_9.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154737063_0.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154737063_2.jpg: 2 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154744341_10.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708154744341_4.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155400789_3.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155400789_4.jpg: 2 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155403184_5.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155405462_0.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155405462_6.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155407878_0.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155407878_5.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155414035_5.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155416419_2.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155418917_11.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155418917_3.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155418917_6.jpg: 1 duplicate labels removed
train: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\train\Image_20240708155418917_7.jpg: 1 duplicate labels removed
train: Scanning D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\labels\train.cache... 1889 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1889/1889 [00:00<?, ?it/s]
D:\LYS\software\anaconda3\envs\yolov12\Lib\site-packages\albumentations\__init__.py:28: UserWarning: A new version of Albumentations is available: '2.0.5' (you have '2.0.4'). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
D:\LYS\pycharmcode\yolov12-main\ultralytics\data\augment.py:1853: UserWarning: Argument(s) 'quality_lower' are not valid for transform ImageCompression
  A.ImageCompression(quality_lower=75, p=0.0),
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
D:\LYS\software\anaconda3\envs\yolov12\Lib\site-packages\albumentations\__init__.py:28: UserWarning: A new version of Albumentations is available: '2.0.5' (you have '2.0.4'). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
D:\LYS\software\anaconda3\envs\yolov12\Lib\site-packages\albumentations\__init__.py:28: UserWarning: A new version of Albumentations is available: '2.0.5' (you have '2.0.4'). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
D:\LYS\software\anaconda3\envs\yolov12\Lib\site-packages\albumentations\__init__.py:28: UserWarning: A new version of Albumentations is available: '2.0.5' (you have '2.0.4'). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
val: Scanning D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\labels\val.cache... 473 images, 0 backgrounds, 0 corrupt: 100%|██████████| 473/473 [00:00<?, ?it/s]
val: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\val\Image_20240708154727009_3.jpg: 1 duplicate labels removed
val: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\val\Image_20240708154746740_1.jpg: 1 duplicate labels removed
val: WARNING ⚠️ D:\LYS\pycharmcode\yolov12-main\dataset\HF2-20141009\images\val\Image_20240708155418917_2.jpg: 2 duplicate labels removed
D:\LYS\software\anaconda3\envs\yolov12\Lib\site-packages\albumentations\__init__.py:28: UserWarning: A new version of Albumentations is available: '2.0.5' (you have '2.0.4'). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
Plotting labels to runs\train\train4\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.001111, momentum=0.9) with parameter groups 131 weight(decay=0.0), 138 weight(decay=0.0005), 137 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 4 dataloader workers
Logging results to runs\train\train4
Starting training for 200 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/200      18.8G      4.748      4.865      2.665         23        640: 100%|██████████| 60/60 [00:19<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.60it/s]
                   all        473      18633   1.58e-05   1.97e-05    8.1e-06   2.43e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/200      18.7G      4.775      4.886      2.662         55        640: 100%|██████████| 60/60 [00:18<00:00,  3.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.55it/s]
                   all        473      18633   5.18e-05   0.000118   2.63e-05   6.58e-06
  0%|          | 0/60 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      3/200      18.8G       4.73      4.874      2.649        110        640: 100%|██████████| 60/60 [00:17<00:00,  3.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.52it/s]
                   all        473      18633    0.00011   0.000337   5.82e-05    1.3e-05
  0%|          | 0/60 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      4/200      18.8G       4.76       4.86      2.656         24        640: 100%|██████████| 60/60 [00:17<00:00,  3.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.53it/s]
                   all        473      18633   0.000133   0.000417   7.06e-05   1.58e-05
  0%|          | 0/60 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      5/200      18.8G      4.754      4.849      2.662         74        640: 100%|██████████| 60/60 [00:17<00:00,  3.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.57it/s]
                   all        473      18633   0.000129   0.000891    6.8e-05   1.45e-05
  0%|          | 0/60 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      6/200      18.7G      4.729      4.844      2.667         30        640: 100%|██████████| 60/60 [00:17<00:00,  3.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.54it/s]
                   all        473      18633   0.000137   0.000913   7.21e-05   1.23e-05
  0%|          | 0/60 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      7/200      18.6G      4.762      4.852      2.667         38        640: 100%|██████████| 60/60 [00:17<00:00,  3.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.44it/s]
                   all        473      18633   0.000123   0.000872   6.62e-05   1.39e-05
  0%|          | 0/60 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      8/200      18.9G      4.749      4.861      2.662        115        640: 100%|██████████| 60/60 [00:17<00:00,  3.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.48it/s]
                   all        473      18633   0.000104   0.000813   5.59e-05   1.09e-05`
